feature/user_profile/presentation/src/main/java/com/sanaa/presentation/screen/watchingHistory/WatchingHistoryScreen.kt

```diff
@@ -56,133 +61,161 @@ fun WatchingHistoryScreen(
-                        if (effect.mediaTypeUi == MediaTypeUi.MOVIE) StartRoute.MOVIE else StartRoute.SERIES
-                    detailsApi.launch(
-                        context = navController.context,
-                        id = effect.id,
-                        startRoute = startRoute
-                    )
+    WatchingHistoryScreenEffectsHandler(
+        effects = viewModel.effect,
+        onNavigateBack = { navController.popBackStack() },
+        onNavigateToMediaDetails = { id, type ->
+            detailsApi.launch(
+                context = navController.context,
+                id = id,
+                startRoute = when (type) {
+                    MediaTypeUi.MOVIE -> StartRoute.MOVIE
+                    MediaTypeUi.TV_SHOW -> StartRoute.SERIES
                 }
-
-                is WatchingHistoryScreenEffect.NavigateBack -> {
-                    navController.popBackStack()
-                }
-            }
-        }
-    }
+            )
+        },
+        onShowErrorSnackBar = { message -> snack = SnackData(message = message, isError = true) },
+        onShowSuccessSnackBar = { message -> snack = SnackData(message = message, isError = false) }
+    )
 
     WatchingHistoryScreenContent(
-        state = state.value,
+        state = state,
         interactionListener = viewModel,
         modifier = modifier,
-    )
-    NovixAnimatedSnackBarHost(
-        data = snack, onDismiss = { snack = null })
-    state.value.selectedMediaToSave?.let { mediaItem ->
-        SaveToListBottomSheet(
-            isVisible = state.value.showSaveToListBottomSheet,
-            mediaId = mediaItem.id.toLong(),
-            onDismiss = viewModel::onDismissSaveToListBottomSheet,
-            onCreateNewListClick = viewModel::onCreateNewListClick,
-            onSuccess = {
-                snack = SnackData(
-                    message = "Added to list successfully",
-                    isError = false
-                )
-            },
-            onFailure = {
-                snack = SnackData(
-                    message = "Added to list failed",
-                    isError = true
-                )
-            },
-        )
-    }
-
-    AddBookmarkListBottomSheet(
-        isVisible = state.value.showAddListBottomSheet,
-        onDismiss = viewModel::onDismissAddListBottomSheet,
-        mediaId = state.value.selectedMediaToSave?.id ?: 0
+        snack = snack,
+        onDismissSnack = { snack = null }
     )
 }
 
+@Composable
+private fun WatchingHistoryScreenEffectsHandler(
+    onNavigateBack: () -> Unit,
+    onNavigateToMediaDetails: (Int, MediaTypeUi) -> Unit,
+    onShowErrorSnackBar: (String) -> Unit,
+    onShowSuccessSnackBar: (String) -> Unit,
+    effects: SharedFlow<WatchingHistoryScreenEffect>
+) {
+    LaunchedEffect(Unit) {
+        effects.collectLatest { effect ->
+            when (effect) {
+                is WatchingHistoryScreenEffect.NavigateBack -> onNavigateBack()
```

Comment:
You can import this as a static import 
```suggestion
                is NavigateBack -> onNavigateBack()
```